session/pingpong/hermes_channel_repository.go

```diff
@@ -252,6 +274,74 @@ func (hcr *HermesChannelRepository) handleNodeStart(payload nodeEvent.Payload) {
 	hcr.fetchKnownChannels(config.GetInt64(config.FlagChainID))
 }
 
+func (hcr *HermesChannelRepository) handleIdentityUnlock(payload identity.AppEventIdentityUnlock) {
+	hermes, err := hcr.addressProvider.GetActiveHermes(payload.ChainID)
+	if err != nil {
+		log.Err(err).Msg("failed to get active Hermes")
+		return
+	}
+	hermesChannel, err := hcr.Fetch(payload.ChainID, payload.ID, hermes)
+	if err != nil {
+		log.Err(err).Msg("failed to get hermes channel info")
+		return
+	}
+
+	unsettledBalance := hermesChannel.UnsettledBalance()
+	if unsettledBalance.Cmp(big.NewInt(0)) == 0 {
```

Comment:
minor things but you could invert this if to `!= 0` and put a `return` in it. That way you wouldn't have to nest this big workflow inside of an `if`.